### PR TITLE
[ext] Refactor type definitions to common

### PR DESCRIPTION
**Pull request by zerefwayne**
This PR introduces a `common` folder to share types between the app and extension projects. This folder will be imported by `diagnostics-app` and `diagnostics-extension` using the `paths` in the respective `tsconfig.json` files.

### CODE REVIEW

**Review by Lamzin — APPROVED** [review]
LGTM, thanks!

File: diagnostics-extension/tsconfig.json [hunk 1]
```diff
@@ -20,5 +21,10 @@
   "exclude": [
     "./node_modules/**/*",
     "./src/__tests__/**/*"
+  ],
+  "references": [
+    {
+      "path": "../common",
```
Comment (MahmoudAGawad) [inline diagnostics-extension/tsconfig.json]: let's also add `"prepend": true`

File: common/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
```
Comment (MahmoudAGawad) [inline common/index.d.ts]: index.d.ts -> index.ts?

File: diagnostics-app/tsconfig.json [hunk 3]
```diff
@@ -26,5 +26,10 @@
     "strictInjectionParameters": true,
     "strictInputAccessModifiers": true,
     "strictTemplates": true
-  }
-}
+  },
+  "references": [
+    {
+      "path": "../common",
```
Comment (MahmoudAGawad) [inline diagnostics-app/tsconfig.json]: let's also add `"prepend": true`

**Review by MahmoudAGawad — COMMENTED** [review]
Overall looks good. One meta comment: Can we add a build step "tsc -b" in each of the projects? This way we make sure that the project is built correctly and included its referenced projects correctly as well. WDYT?

Additionally, there are some guidance on structuring the tsconfig files ... etc, see this https://www.typescriptlang.org/docs/handbook/project-references.html#overall-structure We can just add a bug/issue to track this without the need to do this right now as that is refactoring work. WDYT?

File: common/index.d.ts (hunk 2 above)
Comment (zerefwayne) [reply common/index.d.ts]: I'm getting the same files not inside "rootDir" error if I rename them from `.d.ts` => `.ts`

One possible explanation: .d.ts files are type definitions only used to convey types and are not transpiled to .js files.

**Review by MahmoudAGawad — APPROVED** [review]
LGTM. Looks much better now!

**Pull request merged**